core/hash.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	lib "github.com/NethermindEth/crypto-cpp"
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+// Implements the [Pedersen hash]
+//
+// [Pedersen hash]: https://docs.starknet.io/documentation/develop/Hashing/hash-functions/#pedersen_hash
```

Comment:
Aside: would "PedersenArrayHash" be a more accurate and consistent name for the PedersenDigest function? I thought digest = hash, so PedersenDigest and PedersenHash are synonymous, but currently denote two different functions.

I think the comments on the two Pedersen functions are swapped. If not, please feel free to ignore and resolve these suggestions.
```suggestion
// PedersenDigest implements [Pedersen array hashing].
//
// [Pedersen array hashing]: https://docs.starknet.io/documentation/develop/Hashing/hash-functions/#pedersen_hash
```

Reply:
Noticed that `ArrayDigest` was used in the test function [here](https://github.com/NethermindEth/juno/pull/436/files#diff-ea14ff3f44475668d0756a9a69bfc7f9ac63deef09948915436e3fbc903feaeeR40). We can either go with that or change both to `PedersenArrayHash`. But I also think `PedersenArrayHash` is more appropriate.